src/realm/object-store/object.cpp

```diff
@@ -85,48 +86,64 @@ ModifyPrimaryKeyException::ModifyPrimaryKeyException(const std::string& object_t
 {
 }
 
-Object::Object(SharedRealm r, ObjectSchema const& s, Obj const& o)
-    : m_realm(std::move(r))
-    , m_object_schema(&s)
-    , m_obj(o)
+template <typename Key>
```

Comment:
The refactoring here is to make the big pile of different constructors delegate to a single one which has the new bit of audit logic.